manifests/plugin/remote_execution.pp

```diff
@@ -1,7 +1,57 @@
+# = Foreman Remote Execution
 # Installs foreman_remote_execution plugin
-class foreman::plugin::remote_execution {
+#
+# === Parameters:
+#
+# $cockpit:: Install cockpit support
+#
+class foreman::plugin::remote_execution (
+  Boolean $cockpit = false,
+) {
   include ::foreman::plugin::tasks
+  include apache::mod::proxy_wstunnel
+  include apache::mod::proxy_http
 
   foreman::plugin {'remote_execution':
   }
+
+  $cockpit_subpath = 'webcon'
+
+  if $cockpit {
+    if $::osfamily != 'RedHat' {
+      fail("${::hostname}: foreman_remote_execution cockpit integration does not support osfamily ${::osfamily}")
+    }
+
+    foreman::plugin { 'remote_execution-cockpit': }
+    -> service { 'foreman-cockpit':
+      ensure => running,
+      enable => true,
+    }
+
+    file { '/etc/foreman/cockpit/cockpit.conf':
```

Comment:
If the package creates `/etc/foreman/cockpit` then it also needs a require on `Foreman::Plugin['remote_execution-cockpit']`. Otherwise you need to ensure the directory exists.